### PR TITLE
Document slack channel creation for other projects

### PR DESCRIPTION
**Pull request by castrojo**
Sometimes there's some confusion as to when/how/if a project should ask for a slack channel on the K8s slack. We never really wrote that down so I took some of the excellent recommendations from @jdumars from this issue and wrote them down: https://github.com/kubernetes/community/pull/5107

I'm adding a hold since I'm sure I've missed some of the other things we look for in channel creation to give other people a chance to weigh in. 

/hold
/cc @jdumars 
/sig contributor-experience

### CODE REVIEW

**Review by markyjackson-taulia — APPROVED** [review]
/lgtm

File: communication/slack-guidelines.md [hunk 1]
```diff
@@ -111,6 +110,32 @@ and describe the situation.
...
+  - Establishing and maintaining a Slack channel should be an inflection point
+    in a project's adoption.
+  - Requesting a channel means maintaining it on behalf of the project filing
+    the issue, you will be expected to participate and foster a healthy
+    discourse.
+- Channels around commercial services built on OSS projects are allowed.
+  - Users love the value of being able to collaborate around various services.
+  - Keep it classy, on the Kubernetes Slack we're all on the same team.
+- Ask in #slack-admins or file an issue if you're unsure, it never hurts to ask.
+
```
Comment (coderanger) [inline communication/slack-guidelines.md:138]: I would add:

```
- The channel MUST be public.
  - In order to ensure all channels adhere to our Code of Conduct, we heavily restrict private channels.
  - We do allow `#...-dev` channels for established projects where a single project channel is too noisy, but please don't create both at the start.
  - If you need private discussion areas for security-sensitive topics, a project-specific Slack or the [CNCF Slack] may be a better fit.
```

Comment (mrbobbytables) [reply communication/slack-guidelines.md:138]: +1 to those suggestions. 👍 

Otherwise lgtm

Comment (castrojo) [reply communication/slack-guidelines.md:145]: Thank you so much for not just reviewing but giving me something copy and pasteable, <3.

**Comment by irvifa** [comment]
Wondering why the test keep failing

**Comment by nikhita** [comment]
@irvifa 

```
 Verifying verify-spelling.sh
communication/slack-guidelines.md:123:64: "accomodate" is a misspelling of "accommodate"
communication/slack-guidelines.md:126:68: "concensus" is a misspelling of "consensus"
exit status 2
FAILED   verify-spelling.sh	6s 
```

File: communication/slack-guidelines.md [hunk 2]
```diff
@@ -111,6 +110,39 @@ and describe the situation.
...
+The primary purpose of the Kubernetes slack is for the coordination of the
+Kubernentes project. However it is useful for developers and users to have a
+strong ecosystem of channels for related things. Here are some guidelines for
+determining if you should request a channel:
+
+- The channel MUST be Kubernetes related in some way.
+  - Related cloud native projects might be more appropriate on the [CNCF Slack].
+- The project MUST be open source. 
+  - If you are open sourcing a project DO THAT FIRST, we cannot accomodate every
```
Comment (irvifa) [inline communication/slack-guidelines.md:123]:
```suggestion
  - If you are open sourcing a project DO THAT FIRST, we cannot accommodate every
```

File: communication/slack-guidelines.md [hunk 3]
```diff
@@ -111,6 +110,39 @@ and describe the situation.
...
+    organization's open sourcing launch plans.
+  - The purpose of Slack is to organize an existing community, not seed new ones.
+  - Moderators look at contributor activity, adoption and community concensus
```
Comment (irvifa) [inline communication/slack-guidelines.md:126]:
```suggestion
  - Moderators look at contributor activity, adoption and community consensus
```

**Review by irvifa — CHANGES_REQUESTED** [review]

**Comment by irvifa** [comment]
Ah I see, so is it better to wait until we get success in the test?

**Comment by mrbobbytables** [comment]
I think everything is good - @castrojo do you want to release the hold?

**Pull request merged**